src/Explorer/Tabs/DocumentsTab.ts

```diff
@@ -369,36 +374,22 @@ export default class DocumentsTab extends TabsBase {
     return true;
   };
 
-  public onApplyFilterClick(): Q.Promise<any> {
+  public async refreshDocumentsGrid(): Promise<void> {
     // clear documents grid
     this.documentIds([]);
-    return this.createIterator()
-      .then(
-        // reset iterator
-        iterator => {
-          this._documentsIterator = iterator;
-        }
-      )
-      .then(
-        // load documents
-        () => {
-          return this.loadNextPage();
-        }
-      )
-      .then(() => {
-        // collapse filter
-        this.appliedFilter(this.filterContent());
-        this.isFilterExpanded(false);
-        const focusElement = document.getElementById("errorStatusIcon");
-        focusElement && focusElement.focus();
-      })
-      .catch(error => {
-        window.alert(getErrorMessage(error));
-      });
-  }
 
-  public refreshDocumentsGrid(): Q.Promise<any> {
-    return this.onApplyFilterClick();
+    try {
+      // reset iterator
+      this._documentsIterator = this.createIterator();

```

Comment:
Do you need to `await` here to match old behavior?

Reply:
Nop, getting the iterator is actually a synchronous function. The promise in the old behavior is unnecessary. Fetching documents using the iterator is async however.